addons/website_sale/models/product_template.py

```diff
@@ -190,14 +191,32 @@ def _get_sales_prices(self, pricelist):
                 'price_reduce': price_reduce
             }
 
-            if show_discount:
+            has_compare_list_price = template.compare_list_price
+            has_pricelist_price = pricelist_contains_templates[template.id]
+
+            if has_compare_list_price:
```

Comment:
No need for a variable to store that;
```suggestion
            has_pricelist_price = pricelist_contains_templates[template.id]

            if template.compare_list_price:
```